gateway/src/apicast/executor.lua

```diff
@@ -37,6 +37,26 @@ local function build_context(executor)
     return linked_list.readwrite({}, config)
 end
 
+local function store_original_request(context)
+  -- There are a few phases[0] that req and var are not set and API is disabled.
+  -- The reason to call this using a pcall function is to avoid to define the
+  -- phases manually that are error prone. Also in openresty is not a good method
+  -- to check this. [1]
+  -- [0] invalid phases: init_worker, init, timer and ssl_cer
+  -- [1] https://github.com/openresty/lua-resty-core/blob/9937f5d83367e388da4fcc1d7de2141c9e38d7e2/lib/resty/core/request.lua#L96
+
+  local result, _ =  pcall(function()
```

Comment:
Should we have an `if` here that sets `context.original_request` only when it's nil?
Otherwise, we might overwrite this var when some policy has already modified the headers, or the path. The `shared_build_context` function might guarantee that this never happens, but I'm not 100% sure.

Also, I have doubts about this. You mention that this is going to fail in the `init_worker` phase. However, the `shared_build_context` function will build the context in that phase and store it in `ctx.context`. Subsequent calls to the method will use what's stored in `ctx.context` instead of building again the context, so when is `original_request` initialized?